aggregator/src/datastore.rs

```diff
@@ -2959,6 +2967,18 @@ fn check_single_row_mutation(row_count: u64) -> Result<(), Error> {
     }
 }
 
+/// Add a [`std::time::Duration`] to a [`chrono::NaiveDateTime`].
+fn add_naive_date_time_duration(
+    time: &NaiveDateTime,
+    duration: &StdDuration,
+) -> Result<NaiveDateTime, Error> {
+    time.checked_add_signed(
+        chrono::Duration::from_std(*duration)
+            .map_err(|_| Error::TimeOverflow("overflow converting duration to signed duration"))?,
+    )
+    .ok_or(Error::TimeOverflow("overflow adding duration to time"))
+}
```

Comment:
I can't think of a better place to put this, especially since it's not currently used outside of `datastore.rs`, but I don't love adding unrelated functionality to the already too-large file. However I don't think we can do any better until `chrono` adds better support for `std::time::Duration`. Ah well.